lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -109,9 +129,22 @@ class EngineAutofillForm {
 
     // Validation is in the framework side.
     formElement.noValidate = true;
+    formElement.method = 'post';
+    formElement.action = '#';
+    formElement.addEventListener('submit', (e) {
+      e.preventDefault();
+    });
 
     _hideAutofillElements(formElement);
 
+    // Form elements are always sent with the same order from Framework
+    // therefore no sorting necessary for the ids.
+    final StringBuffer ids = StringBuffer();
```

Comment:
I'm not sure if it's safe to make this assumption. The developer may add/remove items, or reorder the group, no?

Reply:
Reordering should be easy to address with a sort algorithm. Thanks for the suggestion. I'll address it in the next commit.

However, add/remove cases will always cause problem unless Framework sends some additional signals to the engine. This can only be addressed on the Framework level. 

Let me give an example solution algorithm and 3 examples:
(1) **A possible solution**: we can match 2 forms as equal if there are a shared elements.
Previous form: address1 address2 (saved on the DOM due to a previous user edit)
New form: address1 address2 tel email
At this point, when a new `set_client` platform message arrives to the engine, we can identify these forms as equal, first remove the one on the DOM, then add a new  form to the DOM. So looks like this solution can handle some cases, but not all.

(2) _This case will give an error even with the algorithm on 1._
If user fills a formA: address1 address2 tel email , later it's split into 2 parts on the framework such as formB: address1 address2 formC: tel email
If the user clicks on address1 to edit, this will trigger a `set_client` message. According to the algorithm in (1) the engine will remove FormA and will only add info for FormB to the DOM. Information on FormC (Tel email) is no longer attached to the DOM. But visible on the framework. User hits submit button, but browser only saves some part of the info.

(3) _This case will give an error even with the algorithm on 1._
If user fills a formA: address1 address2 tel email , later framework code removes tel email.
If the user never clicks any form elements, no platform messages are send to inform the engine of the change, all four elements are still on the DOM. Only 2 of them are in the framework thus 2 elements are visible to the user. User hits submit button, but browser saves more information than necessary.

There might be multiple ways to address this in the framework:
- Every-time the autofill group elements change (addition/removal), framework can send a finalize message with save=false, therefore. 
- Autofill finalize call, also sends a list of ids of the elements still on the page. Engine creates a new form with the information visible to the user on the time they triggered a finalize event. Only information still on the page is saved to the browser.

I assume other engine's such as desktop which can show larger number of form elements will experience similar issues in the long run, therefore it is better to open an issue and record all the possible problems that can arise from adding/removing elements to an autofill group dynamically.

Reply:
I added the check for reordered element.